decidim-census/app/models/decidim/census/csv_data.rb

```diff
@@ -0,0 +1,33 @@
+require 'csv'
+
+module Decidim
+  module Census
+    class CsvData
+
+      attr_reader :errors, :values
+
+      def initialize(file)
+        @file = file
+        @errors = []
+        @values = []
+
+        CSV.foreach(@file, headers: true, col_sep: ';') do |row|
```

Comment:
For me it's surprising that when I am instantiating an object it performs some kind of action. This is not a mistake. It's just because I am too used to using the form:

```ruby
CsvData.new(file).call
```

However, I imagine that your point is that the object processing should only be run once (with the same file param). So there is no need to have a method call that can be called several times.

It's just a reflexion. I'm not expecting any change here.

Reply:
Yes... Probably `CsvData.new.call(file)`  is better...

Reply:
Now I'm looking into this issue:

> For me it's surprising that when I am instantiating an object it performs some kind of action.

Yes, I understand... but you only know that if you look at the code *inside* the class ;-) In fact, you can treat it like a kind of object initialization...

> I imagine that your point is that the object processing should only be run once

Yes. Exactly. 

Probably `CsvData.new(file).call` it's more idiomatic, but I'm afraid to have an object instance in an inconsistency state... What if you forget to call `call`?